Content.Server/Xenoarchaeology/XenoArtifacts/Effects/Systems/PortalArtifactSystem.cs

```diff
@@ -27,12 +27,14 @@ private void OnActivate(Entity<PortalArtifactComponent> artifact, ref ArtifactAc
         for (int i = 0; i < 50; i++)
         {
             var rndCheck = _random.Pick(mindQuery);
-            if (rndCheck.HasMind)
-            {
-                target = rndCheck;
-                break;
-            };
+
+            if (!rndCheck.HasMind) continue;
+            if (Transform(rndCheck.Owner).MapID != Transform(artifact).MapID) continue;
+
+            target = rndCheck;
+            break;
         }
+    }
         if (target == null) return;
```

Comment:
i think this stuff should be replaced with EQE<MindContainerComponent, TransformComponent> and then filter with `.Where((uid, mc, xform) => mc.HasMind && xform.MapID == map).ToList()` and pick it once after. if the list is empty just return
get artifact map once *before* the filtering since it wont change `var map = Transform(artifact).MapID);`

Reply:
Sloth swears a lot every time he sees me using LINQ
🤔

Reply:
already imported linq and how you do this with 0 < 50 is highly evil

Reply:
non-linq would be to create a list, foreach the enumerator, add if it matches the condition
then pick that

which would still be better but idk if they compile to same thing

Reply:
it turns out that the anomaly generation code is evil

Reply:
Ok

Reply:
![image](https://github.com/space-wizards/space-station-14/assets/96445749/db7309de-43ac-430d-8c3c-9fcdc0569b35)